src/chains/ethereum/ethereum/src/miner/executables.ts

```diff
@@ -1,7 +1,42 @@
 import { TypedTransaction } from "@ganache/ethereum-transaction";
 import { Heap } from "@ganache/utils";
+import { byNonce } from "../transaction-pool";
 
-export type Executables = {
-  inProgress: Set<TypedTransaction>;
-  pending: Map<string, Heap<TypedTransaction>>;
-};
+export class Executables {
+  public inProgress: Set<TypedTransaction> = new Set();
+  public pending: Map<string, Heap<TypedTransaction>> = new Map();
+
+  /**
+   * Deep clones the executables, moving all `inProgress` transactions to the
+   * pending queue and unlocking all transactions.
+   * @returns Cloned and reset executables.
+   */
+  public cloneAndReset() {
+    const executables = new Executables();
+    const { inProgress, pending } = this;
+
+    inProgress.forEach(transaction => {
+      const copy = transaction.copy();
+      copy.locked = false;
+      const origin = copy.from.toString();
+      const txsFromOrigin = executables.pending.get(origin);
+      if (txsFromOrigin) txsFromOrigin.push(copy);
+      else executables.pending.set(origin, Heap.from(copy, byNonce));
```

Comment:
Erg - I'm not a huge fan of this if/else styling from prettier

Reply:
Yeah I don't love it either, and prettier doesn't force it. Fixed.